black.md

```diff
@@ -0,0 +1,74 @@
+- Feature Name: blackened-seasoning
+- Start Date: 2019-10-04
+- SEP Status: Draft
+- SEP PR: (leave this empty)
+- Salt Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Adopt `pre-commit`, `black`, and `isort` for Salt.
+
+# Motivation
+[motivation]: #motivation
+
+There have been a *number* of PRs to Salt that have had minor style changes
+requested, either to bring the coding style in to line with the general Salt
+project or within the specific module, when it deviates from Salt's typical
+standards.
+
+Additionally there are a number of commits that are produced just to satisfy
+the linter. If we automate the formatting of our code, this goes away. We also
+achieve consistency throughout the Salt codebase.    
+
+
+# Design
+[design]: #detailed-design
+
+### Black, Pre-commit, and You
+ 
+Salt will adopt [Black](https://github.com/psf/black) as our code formatter of
+choice. By adopting an opinionated formatter we are taking the approach that
+there is only “one way to do it”. Accepting double quotes was extremely
+difficult for Tom, but if he can do it, anyone can.
+
+Salt will also adopt [pre-commit](https://pre-commit.com) in order to automate
+the use of Black and lint. As `pre-commit` runs `black`, `isort` and lint
```

Comment:
I recommend [https://github.com/asottile/reorder_python_imports](reorder_python_imports) instead of isort. It shares black's philosophy of being nearly unconfigurable.

If you want to use isort, I would also suggest using reorder-python-imports's choice of breaking imports into individual lines by setting `force_single_line` to `True`. This will achieve [reorder-python-imports's stated single aim](https://github.com/asottile/reorder_python_imports#why-this-style) of having a practical effect beyond just style - avoiding merge conflicts. With the number of Salt PRs, that could really help.

Reply:
Reorder python imports does not account for PyLint comments, inline or before the import, it just leaves them behind or it would also be my choice.

But yes, I agree with you, one import per line to reduce conflicts.

Reply:
Good point, thanks.

Reply:
I think that's fair enough. While `black` helps when you've got a lot of imports, simply forcing them to one line per import should be enough.

And given the automated nature of the changes it shouldn't really be a big deal - write your imports how you will, and they'll get updated by the tools 👍